LeopardCommon.h

```diff
@@ -186,6 +186,11 @@
 // Unroll inner loops 4 times
 #define LEO_USE_VECTOR4_OPT
 
+// MacOS M1
+#if defined(__aarch64__)
+  #define LEO_USE_SSE2NEON
+  #define LEO_TARGET_MOBILE
```

Comment:
Does `LEO_TARGET_MOBILE` hold for the general case, e.g. [Mac Studio](https://www.apple.com/shop/buy-mac/mac-studio) with M1 MAX or M1 ULTRA?

Reply:
`LEO_TARGET_MOBILE` is a misnomer. It basically means  "non-Intel and ARM". In the original code it was turned on by ANDROID and IOS defines, and it serves to disable checks for Intel features and enable some ARM specific code.

I chose not to rename it to minimize the impact of my patch.

Based on my checks on different architectures I would say it will most probably work on those Macs as well.